### DISCUSSION TITLE
How to play from multiple speakers using the playAudio Method

### DISCUSSION THREAD

**Discussion by daikyoukai110** [cc-tweaked/CC-Tweaked#1635] [Q&A]
Hello. I am asking this question because I have run into a little bit of an obstacle that I cannot find the answer to myself.

As the title says, I would like to play audio from multiple speakers on Rednet at the same time using playAudio mesos.
When using the playNote mesosot, I was able to play music without delay regardless of wired or wireless connection. However, when using the playAudio mesosot, only one speaker at a time can play the music, which causes a lag, which is not good.
Looking at the code, I think os.pullEvent("speaker_audio_empty") is affecting it, so I tried using parallel.waitForAll, but that didn't help.
I would appreciate it if you could help me.

**Comment by fatboychummy** [comment]
Each speaker queues its own `speaker_audio_empty` event, so you'll need to filter which speaker has finished and send the relevant data to that speaker.

I'll recommend using something like @MCJack123's [AUKit](https://github.com/MCJack123/AUKit) (and [docs](https://mcjack123.github.io/AUKit/)). It does the part of dealing with the audio buffer and speakers, and works very well.

[I wrote a short writeup on Reddit a while ago on how to use AUKit](https://www.reddit.com/r/ComputerCraft/comments/14wsdl5/comment/jrkryi5/), if you wish to use it.
